relay-config/src/config.rs

```diff
@@ -643,6 +700,160 @@ impl Config {
         Ok(config)
     }
 
+    /// Override configuration with values coming from other sources (e.g. env variables or
+    /// command line parameters)
+    pub fn override_config(
+        &mut self,
+        overrides: OverridableConfig,
+    ) -> Result<&mut Self, ConfigError> {
+        let relay = &mut self.values.relay;
+
+        if let Some(upstream) = overrides.upstream {
+            relay.upstream = UpstreamDescriptor::from_str(&upstream).map_err(|err| {
+                ConfigError::new_field_error(
+                    "upstream",
+                    upstream,
+                    err.context(ConfigErrorKind::InvalidOverride),
+                )
+            })?;
+        }
+
+        if let Some(host) = overrides.host {
+            relay.host = IpAddr::from_str(&host).map_err(|err| {
+                ConfigError::new_field_error(
+                    "host",
+                    host,
+                    err.context(ConfigErrorKind::InvalidOverride),
+                )
+            })?;
+        }
+
+        if let Some(port) = overrides.port {
+            relay.port = u16::from_str_radix(port.as_str(), 10).map_err(|err| {
+                ConfigError::new_field_error(
+                    "port",
+                    port,
+                    err.context(ConfigErrorKind::InvalidOverride),
+                )
+            })?;
+        }
+
+        let processing = &mut self.values.processing;
+        if let Some(enabled) = overrides.processing {
+            match enabled.to_lowercase().as_str() {
+                "true" => processing.enabled = true,
+                "false" => processing.enabled = false,
+                val => {
+                    return Err(ConfigError::new_field_error(
+                        "processing",
+                        val,
+                        Context::new(ConfigErrorKind::InvalidOverride),
+                    ));
+                }
+            }
+        }
+
+        // TODO are we going to support Cluster redis ?
+        if let Some(redis) = overrides.redis_url {
+            processing.redis = Some(RedisConfig::Single(redis))
+        }
+
+        if let Some(kafka_url) = overrides.kafka_url {
+            let mut bootstrap_servers_found = false;
+            for entry in processing.kafka_config.iter_mut() {
+                if entry.name.as_str() == "bootstrap.servers" {
+                    entry.value = kafka_url.clone();
```

Comment:
To avoid this clone, you could use something like this (also, no need to convert to `str`):

```rust
let existing = processing.kafka_config.iter_mut()
  .find(|e| e.name == "bootstrap.servers");

if let Some(entry) = existing {
  entry.value = kafka_url;
} else {
  processing.kafka_config.push(...);
}
```

Reply:
nice